src/openrct2/ride/VehiclePaint.cpp

```diff
@@ -1995,23 +1952,11 @@ static void vehicle_sprite_6_0(
     paint_session& session, const Vehicle* vehicle, int32_t imageDirection, int32_t z,
     const rct_ride_entry_vehicle* vehicleEntry)
 {
-    if (vehicleEntry->sprite_flags & VEHICLE_SPRITE_FLAG_GENTLE_SLOPES)
+    if (vehicleEntry->GroupEnabled(Slopes25))
     {
-        if (vehicleEntry->flags & VEHICLE_ENTRY_FLAG_SPINNING_ADDITIONAL_FRAMES)
-        {
-            int32_t boundingBoxNum = ((YawTo16(imageDirection)) ^ 8) + 16;
-            int32_t spriteNum = (((YawTo4(imageDirection)) + 12) * vehicleEntry->base_num_frames)
-                + vehicleEntry->gentle_slope_image_id;
-            VehicleSpritePaintWithSwinging(session, vehicle, spriteNum, boundingBoxNum, z, vehicleEntry);
-        }
-        else
-        {
-            int32_t boundingBoxNum = ((YawTo16(imageDirection)) ^ 8) + 16;
-            int32_t spriteNum = ((vehicleEntry->SpriteByYaw(imageDirection) + 8 + vehicleEntry->NumRotationFrames())
-                                 * vehicleEntry->base_num_frames)
-                + vehicleEntry->gentle_slope_image_id;
-            VehicleSpritePaintWithSwinging(session, vehicle, spriteNum, boundingBoxNum, z, vehicleEntry);
-        }
+        int32_t boundingBoxNum = ((YawTo16(imageDirection)) ^ 8) + 16;
```

Comment:
i assume you have tested this "change" a bit (Yeah i get that its not a change its just a refactoring moving the work into the loader but we need to verify it still works 100% for both previous branches)

Reply:
It worked when I last looked at the spinning wild mouse (the only car that hit the branch AFAIK) but it was a few commits ago.